### ISSUE TITLE
R0913 excludes keyword- and positional-only arguments

### ISSUE THREAD

**Issue by charliermarsh** [pylint-dev/pylint#8667]
### Bug description

Pylint's `too-many-arguments` rule seems to exclude keyword- and positional-only arguments from the "total argument" count:

```py
def f(a, b, c, d, e, *, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t):
    ...


def g(a, b, c, d, e, f):
    ...
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:5:0: R0913: Too many arguments (6/5) (too-many-arguments)
```

### Expected behavior

My instinct is that it should include keyword- and positional-only arguments, but I'm wondering if this is intentional.

### Pylint version

```shell
pylint 2.17.2
astroid 2.15.3
Python 3.11.1 (main, Dec 23 2022, 09:25:23) [Clang 14.0.0 (clang-1400.0.29.202)]
```

**Comment by mbyrnepr2** [comment]
Thanks for this @charliermarsh! I believe that the `too-many-arguments` checker pre-dates positional-only arguments and explains why they were not originally handled by the checker. I don't think there is any reason to exclude the positional-only and keyword-only parameters from the check.

**Comment by parched** [comment]
@charliermarsh @mbyrnepr2 This going to throw up a lot of warnings in my code. I believe the intention of this rule is to ensure args don't get mixed up. That can't happen with kwargs, as they're explicit. I do believe position-only args should be included though.

**Comment by Pierre-Sassoulas** [comment]
I think that ``too-many-arguments`` is a [refactoring check and its intention is only to limit the number of args of a function](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/too-many-arguments.html) (i.e. you should create a data structure to group semantically close arguments together). We asked ourselves if we should increase the default value from 5 to 7 though, there's indeed a lot of new message raised. Not having too many positional argument because you don't know what is what is another check not yet implemented (https://github.com/pylint-dev/pylint/issues/385)

**Comment by haukex** [comment]
I just ran into this on upgrading to 3.0.0 and indeed did have to increase my `max-args` from 5 to 7. I also agree with @parched's point about this check being useful for avoiding confusion when there are too many *positional* arguments - excluding keyword arguments from the check would a nice thing to have too.

**Comment by sabik** [comment]
Thirded (fourthed?) on the distinction between too many arguments total and too many positional arguments, and submitted as a separate feature request issue #9099
